pkg/query/storeset.go

```diff
@@ -203,13 +219,30 @@ func (s *StoreSet) Update(ctx context.Context) {
 		// Peer does not exists anymore.
 		st.close()
 		delete(s.stores, addr)
-
-		level.Info(s.logger).Log("msg", "store unhealthy or does not exists, closed gRPC client and removed from storeset", "address", addr)
+		s.updateStoreStatus(&StoreStatus{
```

Comment:
Can you pass `store.Client` to updateStoreStatus directly to make this code more compact (since you probably can't pass type `st` there directly)? You can calculate LastCheck (time.Now) inside updateStoreStatus directly.